backend/app/core/middleware.py

```
@@ -0,0 +1,281 @@
+        try:
+            response = await call_next(request)
+            
+            # Calculate response time
+            process_time = time.time() - start_time
+            response.headers["X-Process-Time"] = str(process_time)
+            response.headers["X-Correlation-ID"] = correlation_id
+            
+            # Log successful response
+            await self._log_response(request, response, process_time, correlation_id)
+            
+            return response
```

Ruff (0.14.2)

Consider moving this statement to an `else` block

(TRY300)